From 7bdc368e159c64de5b5aa49af76d6c614130490c Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:51:38 -0400
Subject: [fs] nfsd: getattr for FATTR4_WORD0_FILES_AVAIL needs the statfs buffer

Message-id: <1403560314-32525-157-git-send-email-bfields@redhat.com>
Patchwork-id: 84593
O-Subject: [Patch RHEL7 156/172] nfsd: getattr for FATTR4_WORD0_FILES_AVAIL needs the statfs buffer
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Christoph Hellwig <hch@lst.de>

Note nobody's ever noticed because the typical client probably never
requests FILES_AVAIL without also requesting something else on the list.

Signed-off-by: Christoph Hellwig <hch@lst.de>
Cc: stable@vger.kernel.org
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 12337901d654415d9f764b5f5ba50052e9700f37
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 70d1472..3976dc6 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -2049,8 +2049,8 @@ nfsd4_encode_fattr(struct xdr_stream *xdr, struct svc_fh *fhp,
  err = vfs_getattr(&path, &stat);
  if (err)
   goto out_nfserr;
- if ((bmval0 & (FATTR4_WORD0_FILES_FREE | FATTR4_WORD0_FILES_TOTAL |
-   FATTR4_WORD0_MAXNAME)) ||
+ if ((bmval0 & (FATTR4_WORD0_FILES_AVAIL | FATTR4_WORD0_FILES_FREE |
+   FATTR4_WORD0_FILES_TOTAL | FATTR4_WORD0_MAXNAME)) ||
      (bmval1 & (FATTR4_WORD1_SPACE_AVAIL | FATTR4_WORD1_SPACE_FREE |
          FATTR4_WORD1_SPACE_TOTAL))) {
   err = vfs_statfs(&path, &statfs);
-- 
1.7.1